### ISSUE TITLE
Possible Bug (or Even Hack) in Evaluation

### ISSUE THREAD

**Issue by WxxShirley** [OSU-NLP-Group/TravelPlanner#51]
Dear authors,

I recently came across another potential issue in the evaluation process. Specifically, the exact string match for **"Self-driving"** may cause many other expressions referring to the same mode, e.g., “Drive”, “Driving”, etc., to be overlooked. As a result, **these cases may incorrectly pass commonsense evaluations**, including hallucinations about reachability from source to destination, transportation consistency, and **hard constraints** such as transportation preference satisfaction and total budget.

This phenomenon is less severe for a pre-trained model because its direct inference performance is relatively limited. However, during RL training, **this loose matching leads the agent to exploit the system by intentionally using terms like “Drive” or “Driving”** instead of “Self-driving,” thereby achieving higher rewards. To support this observation, I include training dynamics of tool invocation of a RFT agent, where the agent already adheres to the ReAct paradigm (`<think></think> <tool_call></tool_call>` and `<think></think> <answer></answer>`). We can clearly see that the number of `SearchFlight` tool calls decreases significantly, eventually reaching zero, while `GoogleDistanceMatrix` calls increase.

<img width="379" height="309" alt="Image" src="https://github.com/user-attachments/assets/07a97512-51a1-4bd6-998e-00afc447bd1f" />

<img width="388" height="309" alt="Image" src="https://github.com/user-attachments/assets/983c5aab-813f-478c-954c-6f563306e974" />

<img width="375" height="305" alt="Image" src="https://github.com/user-attachments/assets/27398049-0ca5-49d8-9490-99c0d79d9c15" />

I have already addressed this issue in my own implementation when training agents on TravelPlanner. The transportation-matching function now recognizes **a wider range of expressions referring to "Self-driving"**, and **any case that does not detect "Flight" or "Taxi" is treated as "Self-driving" by default**, resulting in a stricter evaluation. I would like to extend my validation to the official TravelPlanner test set, which depends on your implementation. Therefore, I am wondering whether the authors could kindly apply similar stricter checks for the "Self-driving" mode. I understand you may be very busy, if it is more convenient, would you be open to my submitting a Pull Request? If this is not permitted, I would greatly appreciate your permission to infer the constraints from the test set and perform informal testing using my own implementation.

Finally, I sincerely appreciate the authors’ insightful TravelPlanner benchmark. It has been extremely helpful as a multi-turn tool-use testbed for RL training under the standard ReAct paradigm. Thank you for your valuable contributions to the field, and I apologize for any inconvenience caused by raising this issue.

Best,
Xixi
